--- ./drivers/tty/serial/8250/8250_pci.c	2016-12-11 20:17:54.000000000 +0100
+++ ./drivers/tty/serial/8250/8250_pci.c_new	2017-02-15 13:09:23.318199683 +0100
@@ -1642,13 +1642,13 @@ pci_xr17v35x_setup(struct serial_private
 		writeb(0x00, p + UART_EXAR_MPIOLVL_7_0);
 		writeb(0x00, p + UART_EXAR_MPIO3T_7_0);
 		writeb(0x00, p + UART_EXAR_MPIOINV_7_0);
-		writeb(0x00, p + UART_EXAR_MPIOSEL_7_0);
+	//	writeb(0x00, p + UART_EXAR_MPIOSEL_7_0); // Don't touch deafult direction. Fix for the Brick PC.
 		writeb(0x00, p + UART_EXAR_MPIOOD_7_0);
 		writeb(0x00, p + UART_EXAR_MPIOINT_15_8);
 		writeb(0x00, p + UART_EXAR_MPIOLVL_15_8);
 		writeb(0x00, p + UART_EXAR_MPIO3T_15_8);
 		writeb(0x00, p + UART_EXAR_MPIOINV_15_8);
-		writeb(0x00, p + UART_EXAR_MPIOSEL_15_8);
+	//	writeb(0x00, p + UART_EXAR_MPIOSEL_15_8); // Don't touch default direction. Fix for the Brick PC.
 		writeb(0x00, p + UART_EXAR_MPIOOD_15_8);
 	}
 	writeb(0x00, p + UART_EXAR_8XMODE);
